### PR TITLE
restructure unit test to incrementally add rule files

### PR DESCRIPTION
**Pull request by GregHanson**
this PR restructures the proxy v2 unit tests so that rule files can be incrementally added to the test env

follow-up PR will add `pilot/pkg/proxy/v2/goldens/` dir and populate golden files

### CODE REVIEW

File: tests/util/pilot_server.go [hunk 1]
```diff
@@ -154,8 +163,13 @@ func setup() error {
 				string(serviceregistry.MockRegistry)},
 		},
 	}
-	// Static testdata, should include all configs we want to test.
-	args.Config.FileDir = IstioSrc + "/tests/testdata/config"
```
Comment (rshriram) [inline tests/util/pilot_server.go]: IstioSrc missing?

File: pilot/pkg/proxy/envoy/v2/cds_test.go [hunk 2]
```diff
@@ -24,22 +24,24 @@ import (
...
+		cdsr := connectADS(t, util.MockPilotGrpcAddr)
+		sendCDSReq(t, sidecarId(app3Ip, "app3"), cdsr)
+
+		res, err := cdsr.Recv()
+		if err != nil {
+			t.Fatal("Failed to receive CDS", err)
+			return
+		}
+
+		strResponse, _ := model.ToJSONWithIndent(res, " ")
```
Comment (rshriram) [inline pilot/pkg/proxy/envoy/v2/cds_test.go]: do you think we can use this opportunity to move to YAMLs? Easier to view in GH

**Comment by costinm** [comment]
Please move to master.

**Comment by costinm** [comment]
BTW - adding the golden files is great, I'm objecting to the 'sleep to wait'. How about reading the response from ADS/etc - and using golden files for individual sections. This way more config can be added to the dir, and we can still parallelize the tests.

I'm trying to get a similar configuration for the tests/helm directory, where we should have configs using
all istio features, deploy them in a stable cluster - and be able to run long running tests against a stable config ( plus few dynamic tests that specifically check how fast a change is propagated and that change doesn't result in loss )

**Pull request closed without merging**